app/assets/javascripts/nbp_wallet.js

```diff
@@ -0,0 +1,52 @@
+let templateValidity = 0;
+let intervalID;
+let timeoutID;
+
+const checkStatus = async () => {
+  try {
+    const response = await fetch(Routes.nbp_wallet_relationship_status_users_path());
+    const json = await response.json();
+
+    if (json.status === 'ready' && window.location.pathname === Routes.nbp_wallet_connect_users_path()) {
+      window.location.pathname = Routes.nbp_wallet_finalize_users_path();
+      return;
+    }
+  } catch (error) {
+    console.error(error);
+  }
+  timeoutID = setTimeout(checkStatus, 1000);
+};
+
+const countdownValidity = () => {
+  if (templateValidity > 0) {
+    templateValidity -= 1;
+  }
+  if (templateValidity === 0 && window.location.pathname === Routes.nbp_wallet_connect_users_path()) {
```

Comment:
At least in my tests, `templateValidity === 0` won't ever be true.

For me, 
```js
document.querySelector('[data-id="nbp_wallet_qr_code"]').dataset.remainingValidity
```
returns a float. Since we are always subtracting 1, we will never reach 0.

<img width="1059" alt="Screenshot 2025-02-12 at 13 17 08" src="https://github.com/user-attachments/assets/b60cef5a-cb9f-4aec-8cda-7132031c8c3c" />

Reply:
I accidently dropped the conversion to seconds, will add it back. Thanks for pointing it out!

Reply:
Done